packages/trader/src/Stores/Modules/Trading/Helpers/proposal.ts

```diff
@@ -18,60 +45,69 @@ export const getProposalErrorField = response => {
     return el_error && isVisible(el_error) ? error_id : null;
 };
 
-export const getProposalInfo = (store, response, obj_prev_contract_basis) => {
-    const proposal = response.proposal || {};
-    const profit = proposal.payout - proposal.ask_price || 0;
-    const returns = (profit * 100) / (proposal.ask_price || 1);
-    const stake = proposal.display_value;
+export const getProposalInfo = (
+    store: TTradeStore,
+    response: PriceProposalResponse & TError,
+    obj_prev_contract_basis: TObjContractBasis
+) => {
+    const proposal = response.proposal || ({} as PriceProposalResponse['proposal']);
+    const profit = (proposal?.payout || 0) - (proposal?.ask_price || 0);
+    const returns = (profit * 100) / (proposal?.ask_price || 1);
+    const stake = proposal?.display_value;
     const basis_list = store.basis_list;
 
-    const contract_basis = store.is_vanilla
+    const contract_basis: TObjContractBasis | undefined = store.is_vanilla
         ? { text: 'Payout', value: 'number_of_contracts' }
-        : basis_list.find(o => o.value !== store.basis) || {};
+        : basis_list.find(o => o.value !== store.basis);
 
-    const is_stake = contract_basis.value === 'stake';
-    const price = is_stake ? stake : proposal[contract_basis.value];
-    let has_increased = price > obj_prev_contract_basis.value;
+    const is_stake = contract_basis?.value === 'stake';
+
+    const price = is_stake ? stake : proposal?.[contract_basis?.value as keyof PriceProposalResponse['proposal']];
+    let has_increased = false;
+
+    if (price !== undefined) {
+        has_increased = price > obj_prev_contract_basis.value;
+    }
 
     if (!obj_prev_contract_basis.value || price === obj_prev_contract_basis.value) {
-        has_increased = null;
+        has_increased = !!null;
     }
 
     const obj_contract_basis = {
-        text: contract_basis.text || '',
+        text: contract_basis?.text || '',
         value: price || '',
     };
 
-    const commission = proposal.commission;
-    const cancellation = proposal.cancellation;
+    const commission = proposal?.commission;
```

Comment:
@henry-deriv `?` marks after `proposal` should not be necessary because it cannot be undefined. it will be assigned with `{}` at line 53 if there is no `proposal` in response.

Reply:
removing `?mark` throws ts error. I added it due to that. I just tried again. Same error persists.

Reply:
@henry-deriv which error are you getting?  I've removed all `?` marks and haven't got any error 🙂 :  
<img width="1129" alt="Screenshot 2023-07-05 at 11 14 34 AM" src="https://github.com/kate-deriv/deriv-app/assets/103177211/d64202f4-0e41-423f-a704-68cc75f52bc3">

Reply:
oh weird. It was just there. Not anymore though!